### PR TITLE
Add setAll function.

### PR DESCRIPTION
**Pull request by porfirioribeiro**
Useful when we need to change multiple properties at once.

This probably could be done on `set()` by accepting an object map to it.

### CODE REVIEW

**Comment by zeckdude** [comment]
This is awesome! Please merge!

**Comment by bookcasey** [comment]
Would love to see this merged in.

**Comment by porfirioribeiro** [comment]
What i do now is have it on my project so i can use, but i have to keep coping the file over projects.

I probably should take some time and make a fork of this library with this function included and other's you see in other pull requests

File: README.md [hunk 1]
```diff
@@ -148,6 +148,18 @@ dotProp.set(obj, 'foo.$end', 'platin-unicorn')
 
 ```
 
+### setAll
```
Comment (hilleer) [inline README.md:151]:
```suggestion
### setMany
```

File: index.js [hunk 2]
```diff
@@ -29,6 +29,17 @@ function set(obj, prop, value) {
 	return setPropImmutableRec(obj, prop, value, 0);
 }
 
+/**
+ * Set many values defined in a map on the obj
+ * @param obj
+ * @param map
+ */
+function setAll(obj, map) {
```
Comment (hilleer) [inline index.js:37]:
```suggestion
function setMany(obj, map) {
```

File: index.js [hunk 3]
```diff
@@ -167,6 +178,7 @@ function propToArray(prop) {
 
 module.exports = {
 	set: set,
+	setAll: setAll,
```
Comment (hilleer) [inline index.js:181]:
```suggestion
	setMany,
```

**Comment by hilleer** [comment]
Hi @porfirioribeiro ,

have you seen my previous comment?

**Comment by porfirioribeiro** [comment]
> Hi @porfirioribeiro ,
> 
> have you seen my previous comment?

Sorry, i haven't
I see you changed the base branch, so i just fixed the conflicts, it should work

Sorry, i missed it. I will have a look when i

**Comment by hilleer** [comment]
@porfirioribeiro no worries. I am working on a full set of "many" functions. Will keep you updated :)

**Pull request merged**